src/parser.js

```diff
@@ -1,35 +1,27 @@
-import q from 'q'
-import http from 'q-io/http'
+import Promise from 'bluebird'
+import requestAsync from 'request-promise'
 import validator from 'validator'
 import {default as podcastParser} from 'node-podcast-parser'
 
 class Parser {
 
-  static process(response) {
+  static process(xml) {
 
-    const deferred = q.defer()
-    podcastParser(response, (err, data) => {
-      if (err) {
-        deferred.reject(new Error(err))
-      } else
-        deferred.resolve(data)
+    return new Promise((res, rej) => {
+      podcastParser(xml, (err, data) => err ? rej(err) : res(data))
```

Comment:
Bluebird has an [Promisification API](http://bluebirdjs.com/docs/api/promise.promisify.html) that I think we could use here.